app/services/record_participant_event.rb

```diff
@@ -59,7 +59,7 @@ def actual_lead_provider
   end
 
   def validate_provider!
-    raise ActionController::ParameterMissing, I18n.t(:invalid_participant) unless actual_lead_provider.nil? || lead_provider == actual_lead_provider
+    raise ActionController::ParameterMissing, I18n.t(:invalid_participant) unless actual_lead_provider.nil? || @params[:lead_provider] == actual_lead_provider
```

Comment:
params is defined using an attr_accessor, so the "@" isn't required.
Originally `lead_provider` was a method generated by InitializeWithConfig. May be better to keep it as a method in case we undo the previous refactor that took that out.